src/asset-swapper/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -2044,6 +2062,20 @@ export class SamplerOperations {
                             ...intermediateTokens.map((t) => [takerToken, t, makerToken]),
                         ].map((path) => this.uniswapV3Sampler.createSampleBuysOperation(path, makerFillAmounts));
                     }
+                    case ERC20BridgeSource.KyberElastic: {
+                        // Rebasing tokens lead to a high revert rate.
+                        if (REBASING_TOKENS.has(takerToken) || REBASING_TOKENS.has(makerToken)) {
+                            return [];
+                        }
+                        const { factory, router } = KYBER_ELASTIC_CONFIG_BY_CHAIN_ID[this.chainId];
+                        if (!isValidAddress(router) || !isValidAddress(factory)) {
```

Comment:
This path will never be hit, see comment on constants.ts.

Reply:
I will remove this validation since the constructor of KyberElasticSampler already does a similar check.

Reply:
Can you point me to where this validation is done? I don't see a constructor in KyberElasticSampler.

Reply:
https://github.com/0xProject/0x-api/blob/19876e1694fabe9dff5a6e6751de45eedc7827f7/src/samplers/kyber_elastic_sampler.ts#L32

your point stands that as is, the constructor validation also won't get hit. i think it's still good to validate the inputs in the constructor in case something is broken in the future